### ISSUE TITLE
[a11y] Refine aria implementation

### ISSUE THREAD

**Issue by steveszc** [cibernox/ember-basic-dropdown#218]
These items will address some a11y issues outlined in [ember-power-select issue #806](https://github.com/cibernox/ember-power-select/issues/806) that are specifically related to the basic-dropdown component. 

- [x] Use aria-owns instead of aria-controls to associate the combobox with the listbox<details>
 https://www.w3.org/TR/wai-aria/roles#combobox
</details>

- [x]  aria-owns/aria-controls must reference an element that exists in the dom.<details>
Currently the trigger's `aria-controls` attribute references an ID of a dom node that does not exist when the power-select is in the collapsed state. This is considered invalid markup and can break screen readers. The controlled element (the listbox) should be maintained in the dom even when collapsed, it can be an empty `div`/`ul`. - https://www.w3.org/TR/wai-aria/states_and_properties#aria-owns
</details>

- [x]  aria-haspopup should not be used on the trigger<details>
`aria-haspopup` is an implicit property of a combobox. Explicitly adding aria-haspopup implicitly and incorrectly defines the element as an aria menu (similar in semantics to a right-click contextual menu or the menu ribbon in MS Word or an OS), which not the correct role for the trigger. This attribute is not necessary on a combobox. 
https://www.w3.org/TR/wai-aria/states_and_properties#aria-haspopup 
https://www.w3.org/TR/wai-aria/roles#combobox
</details>

- [x]  aria-pressed should not be used on the trigger.<details>
`aria-expanded` is sufficient to imply behavior to screen readers. `aria-pressed` is intended for toggle buttons such as checkboxes or tri-state buttons.
https://www.w3.org/TR/wai-aria/states_and_properties#aria-pressed
</details>

**Comment by cibernox** [comment]
> Use aria-owns instead of aria-controls

**Comment by steveszc** [comment]
Per the w3c wai-aria documentation included above, `aria-haspopup` and `aria-pressed` mean something different to a screen reader than a button that expands/collapses some content. For the purposes of basic-dropdown it is sufficient to use `aria-expanded` to indicate the expanded state of the component, and use `aria-owns` to create a parent/child relationship between the two dom nodes (trigger and content). `aria-owns` carries less implicit meaning than `aria-controls` so it's probably a better default for a generic expand/collapse component like basic-dropdown.

**Comment by cibernox** [comment]
Perfect, I'll apply at least some of this feedback this weekend.

**Comment by cibernox** [comment]
All changes applied and released on 0.24.0. The work will continue in ember-power-select
